sopel/modules/ip.py

```diff
@@ -127,9 +128,16 @@ def ip(bot, trigger):
         username = trigger.group(2).strip()
         user_in_botdb = bot.users.get(username)
         if user_in_botdb is not None:
-            query = user_in_botdb.host
+            # Mibbit on not webirc-enabled networks https://wiki.mibbit.com/index.php/Hexip#What_is_shown_when_connected_to_an_IRC_Network_that_is_not_Enabled
+            if user_in_botdb.hostmask.endswith(".mibbit.com"):
+                hexip = user_in_botdb.hostmask.split("!")[1].split("@")[0]
+                # Stolen from https://stackoverflow.com/a/2198052, converts hexip to dotted quad
+                addr_long = int(hexip, 16)
+                query = socket.inet_ntoa(struct.pack(">L", addr_long))
```

Comment:
What does this do for IPv6 addresses? Hex-encoded IPv6 would be too long for the `user` part of the hostmask (max ident length seems to always be 10 characters, and anything past that is simply truncated), so this will only work on IPv4 addresses even if Mibbit tries to send hexip for v6.

Since an IPv4 hexip address should always be 8 characters, I'd say this branch should bail if it's the wrong length. Maybe stick the `query = user_in_botdb.host` assignment before the `if` and drop the `else`?

Reply:
mibbit.com and chat.mibbit.com are not reachable via IPv6

Reply:
Other webirc/CGI:IRC clients might be. See the top-level review comment, which asks why only Mibbit should be supported for hexip lookups.